pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -71,6 +71,24 @@ func (h *HeartBeat) heartBeat(heartBeatInterval time.Duration, stopCh chan struc
 	// 1 minute with a 1.2 jitter + the time it takes for the heartbeat function to run (sliding == true).
 	// So the amount of time between heartbeats randomly varies between 1min and 2min12sec + the heartbeat function execution time.
 	wait.JitterUntil(h.do, heartBeatInterval, 1.2, true, stopCh)
+	now, err := json.Marshal(metav1.Now())
+	if err != nil {
+		log.DefaultLogger().Reason(err).Errorf("Can't determine date")
+		return
+	}
+	cpuManagerEnabled := false
+	kubevirtSchedulable := "false"
+	var data []byte
+	data = []byte(fmt.Sprintf(`{"metadata": { "labels": {"%s": "%s", "%s": "%t"}, "annotations": {"%s": %s}}}`,
```

Comment:
Are you trying to set the schedulable falg to false when they daemonset exists? It is not entirely clear to me what the exact scenario looks like which you try to fix.

Reply:
The basic problem is that there's no logic anywhere to ever remove the "Schedulable" flag once it's set. If virt-handler runs on a node, and it later removed (permanently. e.g. via nodeSelector), the node will forever be marked as schedulable.

Reply:
Hm, that brings up two points:

 1. great to see it addressed
 2. we probably have a  bug in our node controller if it does not go to false after the hearbeat time is over